### PR TITLE
switch to new sync/atomic helpers in go1.19

### PR DESCRIPTION
**Pull request by wadey**
These new helpers make the code a lot cleaner and assert that all references of the field are doing atomically. I confirmed that the
simple helpers like `atomic.Int64` don't add any extra overhead as they
get in-lined by the compiler. `atomic.Pointer` adds an extra method call
as it no longer gets in-lined, but we aren't using these on the hot path
so it is probably okay.

Note that the new go1.19 fmt tool complains about some comments that I also fixed as part of this PR.

No rush to merge this, we may want to wait until go1.19.1 to see if there are any show stopper bugs.

### CODE REVIEW

**Review by brad-defined — COMMENTED** [review]
I like this change a lot. Brings type safety to these atomic values.
I want to wait for Nate to give it a view, so holding off on the 'approve' button for only that reason.

File: connection_manager_test.go [hunk 1]
```diff
@@ -18,6 +18,20 @@ import (
...
+		l:       test.NewLogger(),
+		addrMap: map[iputil.VpnIp]*RemoteList{},
+	}
+	lighthouses := map[iputil.VpnIp]struct{}{}
+	staticList := map[iputil.VpnIp]struct{}{}
+
+	lh.lighthouses.Store(&lighthouses)
+	lh.staticList.Store(&staticList)
+
+	return lh
```
[the remark this answers is not in the document]
Comment (wadey) [reply connection_manager_test.go:32]: I'm not sure if this will work since `atomic.Pointer` has `noCopy` set in the struct (so no copies are allowed, which I think the return from your method would be) https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/sync/atomic/type.go;l=40

[the remark this answers is not in the document]
Comment (brad-defined) [reply connection_manager_test.go:32]: I typed up a local program to use it, and demonstrated to myself (on my M1) that it compiled and executed OK.
I saw that nocopy in the impl, and I dared the compiler to stop me...I posted the note when the compiler failed to prevent my local code snippet, figuring that I wasn't violating anything with that nocopy.

What's going on here? What does that nocopy really mean?
https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/sync/atomic/type.go;l=178;drc=0f6ee42fe063a48d7825bc03097bbb714aafdb7d
leads to a github comment that explains a little more...apparently, it means very little. `go vet` will complain about a copied nocopy value, but nothing else happens. I ran this against my little local test program, and I see `go vet` complaining. So that part at least works.

But why is it wrong to copy an atomic value in the first place? Looks like it should be totally copyable to me. As long as the value is 'moved' (which admittedly doesn't exist as an enforced concept in go) it'd be just fine. Meaning, if there aren't two copies of the value in the world, I can't imagine what could possibly go wrong. Other scenarios I can think of are some underlying atomic implementation is locks under the covers, and locks are also nocopy...which would complicate this analysis. I didn't go searching to see if this was the case.

One answer that fits within this model is to change the atomics to *atomics, but that looks...suboptimal to me. I think there's not enough value in the syntactic sugar of the New method to justify yet another pointer indirection - a pointer to an atomic pointer to an ultimate value.

In conclusion, I can't find anything wrong with the code, but since the suggestion is only syntactic sugar and breaks some `go vet` output (and there may be something I'm missing! atomic code gets tricky very quickly), please do reject the helper function :D

**Review by brad-defined — APPROVED** [review]

**Review by jasikpark — COMMENTED** [review]
looks reasonable to me, who has very little understanding of semantics of atomics
